pkg/validate/configfinder/configfinder.go

```diff
@@ -0,0 +1,40 @@
+package configfinder
+
+import (
+	"context"
+	"github.com/armory/spinnaker-operator/pkg/halconfig"
+)
+
+type ConfigFinder interface {
+	GetAccounts(provider string) (map[string]interface{}, error)
+}
+
+type composedConfigFinder struct {
+	Finders []ConfigFinder
+}
+
+func NewConfigFinder(context context.Context, spinConfig *halconfig.SpinnakerConfig) ConfigFinder {
+	var finders []ConfigFinder
+
+	// NOTE: precedence order is important, last one overwrites previous ones
+	finders = append(finders, &halConfigFinder{SpinConfig: spinConfig, Context: context})
+	finders = append(finders, &profileConfigFinder{SpinConfig: spinConfig, Context: context})
+
+	return &composedConfigFinder{
+		Finders: finders,
+	}
+}
+
+func (f *composedConfigFinder) GetAccounts(provider string) (map[string]interface{}, error) {
+	accounts := map[string]interface{}{}
+	for _, f := range f.Finders {
+		as, err := f.GetAccounts(provider)
+		if err != nil {
+			return nil, err
+		}
+		for name, account := range as {
```

Comment:
This is not how accounts work atm. The last config wins all the accounts. However, I'd like CRD to be additive:

(List1 in `config`, List2 in `clouddriver-local`, List3 as CRDs) -> List2 + List3.

Reply:
Last one wins, but only if the same account name is found in multiple sources. Here the finders are defined in order: https://github.com/armory/spinnaker-operator/pull/25/files/6c11ff601b7617711ab38e633e63b40ff0c8642c#diff-5926d012fef0c048be10851eff01f80dR20
For this example:
```
halconfig:
- kube-1
  ....
- kube-2
  ....
cloud driver-local.yml:
- kube-1
CRD:
- kube-3
  ....
```
The resulting accounts would be:
- kube-1 (from clouddriver-local.yml)
- kube-2 (from halconfig)
- kube-3 (from CRD)

Reply:
AFAIK Spring doesn't reconcile arrays. So in your example, you'd only get `kube-1`. For CRD though, I'd like to allow some accounts defined statically in Spring world + additional ones via CRD.